### DISCUSSION TITLE
First-class SIMD

### DISCUSSION THREAD

**Discussion by taylorpool** [hylo-lang/hylo#1584] [Language design]
Hello,

I am really impressed with Hylo and its vision for mutable value sematics. I think it could have a really strong use case in the low level scene. I think that having first-class SIMD types in the standard library would enable much more low level control by the developer, while also helping out the optimizer. With conditional compilation, this could be pretty ergonomic a la [Mojo](https://github.com/modularml/mojo/blob/main/stdlib/src/builtin/simd.mojo). Is this something that could be possible? Once again, thanks for all your amazing work!

**Comment by dabrahams** [comment]
Thanks for your interest! Sure it could be possible. If you'd like to contribute a design that would help make it a reality. I suggest looking at what Swift has in this area, in addition to Mojo, since Hylo is much closer in design to Swift.

**Reply by taylorpool** [reply]
Thanks for the information!
